contracts/strategies/RFPStrategy.sol

```diff
@@ -0,0 +1,169 @@
+    event Distributed(address indexed recipientId, address recipient, uint256 amount, address token);
+
+    /// ===============================
+    /// ======== Constructor ==========
+    /// ===============================
+
+    constructor(address _allo, string memory _name) BaseStrategy(_allo, _name) {}
+
+    /// ====================================
+    /// =========== Functions ==============
+    /// ====================================
+
+    function initialize(uint256 _poolId, bytes memory _data) public override {
+        super.initialize(_poolId, _data);
+    }
+
+    /// @notice Submit proposal to RFP pool
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the transaction
+    function registerRecipients(bytes memory _data, address _sender) external payable onlyAllo returns (address) {
+        if (payoutReady) {
+            revert READY_FOR_PAYOUT();
+        }
+
+        (address payoutAddress, uint256 proposedAmount, Metadata memory metadata) =
+            abi.decode(_data, (address, uint256, Metadata));
+
+        Recipient memory recipient = Recipient({
+            payoutAddress: payoutAddress,
+            proposedAmount: proposedAmount,
+            recipientStatus: RecipientStatus.Pending
+        });
+
+        address recipientId = address(uint160(_counter++));
+        recipients[recipientId] = recipient;
+
+        emit RecipientRegistered(recipientId, payoutAddress, metadata, _sender);
+
+        return recipientId;
+    }
+
+    /// @notice Checks if msg.sender is eligible for RFP allocation
+    /// @param _recipientId Id of the recipient
+    function getRecipientStatus(address _recipientId) public view override returns (RecipientStatus) {
+        return recipients[_recipientId].recipientStatus;
+    }
+
+    /// @notice Checks address is elgible allocator
+    function isValidAllocator(address _allocator) public view returns (bool) {
+        return allo.isPoolManager(poolId, _allocator);
+    }
+
+    /// @notice Select recipient for RFP allocation
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the allocation
+    function allocate(bytes memory _data, address _sender) external payable onlyAllo nonReentrant {
+        if (!allo.isPoolManager(poolId, _sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        if (payoutReady) {
+            revert READY_FOR_PAYOUT();
+        }
+
+        address approvedRecipientId = abi.decode(_data, (address));
+
+        for (uint256 i = 0; i < _counter;) {
+            address recipientId = address(uint160(i));
+            if (recipientId == approvedRecipientId) {
+                recipients[recipientId].recipientStatus = RecipientStatus.Accepted;
+                acceptedRecipientId = recipientId;
+            }
+
+            recipients[recipientId].recipientStatus = RecipientStatus.Rejected;
+
+            unchecked {
+                i++;
+            }
+        }
```

Comment:
Do we need to update to rejected? It seems worthwhile, but also introduces a loop that could get very long, and no way to get around it to just set the winner to Approved, which is all that's really needed for the rest of the contract to function. 

The better solution I think is to have `getRecipientStatus()` perform this logic:
- if (payoutReady) 
    - if Accepted => Accepted
    - if Pending => Rejected
- else: recipientStatus